sdv/multi_table/hma.py

```diff
@@ -136,21 +144,32 @@ def _augment_table(self, table, tables, table_name):
         """
         self._table_sizes[table_name] = len(table)
         LOGGER.info('Computing extensions for table %s', table_name)
-        for child_name in self.metadata._get_child_map()[table_name]:
+        child_map = self.metadata._get_child_map()[table_name]
+        for child_name in child_map:
             if child_name not in self._augmented_tables:
                 child_table = self._augment_table(tables[child_name], tables, child_name)
 
             else:
                 child_table = tables[child_name]
 
             foreign_keys = self._get_foreign_keys(table_name, child_name)
-            for index, foreign_key in enumerate(foreign_keys):
-                extension = self._get_extension(child_name, child_table.copy(), foreign_key)
+            for foreign_key in foreign_keys:
+                pbar_desc = (
+                    f'({self._learned_relationships + 1}/{len(self.metadata.relationships)}) '
+                    f"Tables '{table_name}' and '{child_name}' ('{foreign_key}')"
+                )
+                extension = self._get_extension(
+                    child_name,
+                    child_table.copy(),
```

Comment:
Is it necessary to copy?

Reply:
Yes because we alter the data and if that child is used in another parent it ends up being altered.